karton/core/backend.py

```diff
@@ -534,6 +534,43 @@ def get_all_tasks(
             self.iter_all_tasks(chunk_size=chunk_size, parse_resources=parse_resources)
         )
 
+    def iter_task_tree(
+        self, root_uid: str, chunk_size: int = 1000, parse_resources: bool = True
+    ) -> Iterator[Task]:
+        """
+        Iterates all tasks that belong to the same analysis task tree
+        and have the same root_uid
+
+        :param root_uid: Root identifier of task tree
+        :param chunk_size: Size of chunks passed to the Redis SCAN and MGET command
+        :param parse_resources: If set to False, resources are not parsed.
+            It speeds up deserialization. Read :py:meth:`Task.unserialize` documentation
+            to learn more.
+        :return: Iterator with task objects
+        """
+        # Process unrouted tasks produced by Karton <5.4.0
+        legacy_task_keys = self.redis.scan_iter(
+            match=f"{KARTON_TASK_NAMESPACE}:*[^:]*", count=chunk_size
+        )
+        for chunk in chunks_iter(legacy_task_keys, chunk_size):
+            yield from filter(
+                lambda task: task.root_uid == root_uid,
+                (
+                    Task.unserialize(task_data, backend=self)
+                    if parse_resources
+                    else Task.unserialize(task_data, parse_resources=False)
```

Comment:
Isn't it just
```suggestion
                    Task.unserialize(task_data, backend=self, parse_resources=parse_resources)
```
?

Reply:
Not entirely, `backend=self` is not passed for one of the cases.

Reply:
Yeah, but on the other hand there's nothing wrong with not passing the backend parameter when parse_resources=False. It's just not required by the unserialize function in that case